### PR TITLE
feat: Add support for replacing all AWS::IAM::Policy with AWS::IAM::ManagedPolicy

### LINKED ISSUE #85: feat: Add support for replacing AWS::IAM::Policy with AWS::IAM::ManagedPolicy
### Description

Some customers don't have permission to attach policies to roles, and must use managed policies. We should add support for replacing the resource type. 

It must remain compatible with AddPermissionBoundary

### Use Case

Customers who cannot deploy standalone policies.

### Proposed Solution

Write an aspect that will identify all `CfnPolicy` objects, and replace them with `CfnManagedPolicy`. The name property needs to be changed. It should also check for any overrides on the PolicyName and apply that change to the ManagedPolicyName

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by ryfestag**
Fixes #85

### CODE REVIEW

File: src/patches/managedPolicies.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { IAspect, aws_iam as iam } from 'aws-cdk-lib';
+import { IConstruct } from 'constructs';
+
+/**
+ * Patch for turning all Policies into ManagedPolicies
+ *
+ * Some customers have policies in place that make it impossible to create standard policies. Instead,
+ * they must use managed policies
+ *
+ * @example
```
Comment (maafk) [inline src/patches/managedPolicies.ts:14]: Can you include a note about how the ordering of Aspects added matters here?

i.e. If used in combination with _other_ Aspects in this project, this should most likely be added _before_ the others are added in.

**Review by maafk — CHANGES_REQUESTED** [review]
Thanks @ryfestag! Just requested a few small changes.

Can you also ensure the `pre-commit` has been run locally ([ref](https://github.com/cdklabs/cdk-enterprise-iac/blob/main/CONTRIBUTING.md#step-4-develop))?  This will append the `API.md` to the bottom of `README.md`

File: src/patches/managedPolicies.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * // Replace all AWS::IAM::Policies with AWS::IAM::ManagedPolicies
+ * Aspects.of(stack).add(new ManagedPolicies())
+ */
+export class ManagedPolicies implements IAspect {
```
[the remark this answers is not in the document]
Comment (ryfestag) [reply src/patches/managedPolicies.ts]: Sure, easy enough. Updated.

File: test/patches/managedPolicies.test.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      runtime: Runtime.PYTHON_3_9,
+    });
+    const bucket = new Bucket(stack, 'TestBucket');
+    bucket.grantReadWrite(func);
+
+    Aspects.of(app).add(new ManagedPolicies());
+    app.synth();
+
+    const appTemplate = Template.fromStack(stack);
+    // Extracted stack has IAM resources
```
[the remark this answers is not in the document]
Comment (ryfestag) [reply test/patches/managedPolicies.test.ts]: Yep, Removed comment.

File: src/patches/managedPolicies.ts (hunk 1 above)
Comment (ryfestag) [reply src/patches/managedPolicies.ts]: Added, let me know if that looks better

**Comment by ryfestag** [comment]
Regarding pre-commit - I needed to tell pre-commit to use the system version. On my end, I needed to specify *something* (either the version of node or tell pre-commit to use the system node). Let me know if that should be done differently

**Review by maafk — APPROVED** [review]

**Pull request merged**